modules/light-clients/08-wasm/types/client_state_test.go

```diff
@@ -352,30 +334,22 @@ func (suite *TypesTestSuite) TestInitialize() {
 
 			tc.malleate()
 
-			var err error
-			initialize := func() {
-				err = clientState.Initialize(suite.ctx, suite.chainA.Codec, clientStore, consensusState)
-			}
-
-			switch {
-			case tc.expPanic != nil:
-				suite.Require().PanicsWithValue(tc.expPanic, initialize)
-			case tc.expError != nil:
-				initialize()
-
-				suite.Require().ErrorIs(err, tc.expError)
-				suite.Require().False(clientStore.Has(host.ClientStateKey()))
-				suite.Require().False(clientStore.Has(host.ConsensusStateKey(clientState.GetLatestHeight())))
-			default:
-				initialize()
+			err := clientState.Initialize(suite.chainA.GetContext(), suite.chainA.Codec, clientStore, consensusState)
 
+			expPass := tc.expError == nil
+			if expPass {
 				suite.Require().NoError(err)
 
 				expClientState := clienttypes.MustMarshalClientState(suite.chainA.Codec, clientState)
-				suite.Require().Equal(clientStore.Get(host.ClientStateKey()), expClientState)
+				suite.Require().Equal(expClientState, clientStore.Get(host.ClientStateKey()))
 
 				expConsensusState := clienttypes.MustMarshalConsensusState(suite.chainA.Codec, consensusState)
-				suite.Require().Equal(clientStore.Get(host.ConsensusStateKey(clientState.GetLatestHeight())), expConsensusState)
+				suite.Require().Equal(expConsensusState, clientStore.Get(host.ConsensusStateKey(clientState.GetLatestHeight())))
+			} else {
+				suite.Require().ErrorIs(err, tc.expError)
+
+				suite.Require().False(clientStore.Has(host.ClientStateKey()))
+				suite.Require().False(clientStore.Has(host.ConsensusStateKey(clientState.GetLatestHeight())))
```

Comment:
as a side note, I think state checks on failed conditions are unnecessary for the same reasons we aren't doing panic checking (baseapp reverts state and that's an assumption we can rely on)